number_parser/parser.py

```diff
@@ -82,24 +83,25 @@ def number_builder(token_list):
             current_grp_value = 0
             previous_token = None
 
-        if (token in BASE_NUMBERS):
-            current_grp_value += BASE_NUMBERS[token]
+        if (token in ALL_BASE_WORDS):
+            current_grp_value += ALL_BASE_WORDS[token]
 
         elif (token in MTENS):
             current_grp_value += MTENS[token]
 
-        elif (token in HUNDRED):
-            if current_grp_value == 0:
-                current_grp_value = 1
-            current_grp_value *= 100
+        elif (token in MHUNDREDS):
+            current_grp_value += MHUNDREDS[token]
 
         elif (token in MULTIPLIERS):
             if current_grp_value == 0:
                 current_grp_value = 1
 
-            current_grp_value *= MULTIPLIERS[token]
-            total_value += current_grp_value
-            current_grp_value = 0
+            if MULTIPLIERS[token] == 100:
+                current_grp_value *= MULTIPLIERS[token]
+            else:
+                current_grp_value *= MULTIPLIERS[token]
+                total_value += current_grp_value
+                current_grp_value = 0
 
         previous_token = token
 
```

Comment:
why is this function returning a list? :thinking: You are adding only one element to the list (`value_list.append(str(total_value))`), so you can just return a single value (the build number).

Reply:
Doing this will simplify the `parse()` and the `parse_number()` functions.

Reply:
Multiple elements could be added into the list too in case the token is invalid. (```if not valid:``` condition in this function) Like in the case of 'one two three' it adds all three numbers (1,2,3) to the list.